skfda/preprocessing/dim_reduction/projection/_fpca.py

```diff
@@ -1,33 +1,433 @@
+            the passed FDataBasis object.
+        component_values_ (array_like): this contains the values (eigenvalues)
+            associated with the principal components.
+        pca_ (sklearn.decomposition.PCA): object for PCA.
+            In both cases (discretized FPCA and basis FPCA) the problem can be
+            reduced to a regular PCA problem and use the framework provided by
+            sklearn to continue.
+
+    Examples:
+        Construct an artificial FDataBasis object and run FPCA with this object.
+        The resulting principal components are not compared because there are
+        several equivalent possibilities.
+
+        >>> data_matrix = np.array([[1.0, 0.0], [0.0, 2.0]])
+        >>> sample_points = [0, 1]
+        >>> fd = FDataGrid(data_matrix, sample_points)
+        >>> basis = skfda.representation.basis.Monomial((0,1), n_basis=2)
+        >>> basis_fd = fd.to_basis(basis)
+        >>> fpca_basis = FPCABasis(2)
+        >>> fpca_basis = fpca_basis.fit(basis_fd)
 
     """
-    fdatagrid = fdatagrid - mean(fdatagrid)  # centers the data
-    # singular value decomposition
-    u, s, v = np.linalg.svd(fdatagrid.data_matrix)
-    principal_directions = v.T  # obtain the eigenvectors matrix
-    eigenvalues = (np.diag(s) ** 2) / (fdatagrid.n_samples - 1)
-    scores = u @ s  # functional principal scores
-
-    return scores, principal_directions, eigenvalues
+
+    def __init__(self,
+                 n_components=3,
+                 components_basis=None,
+                 centering=True,
+                 regularization_derivative_degree=2,
+                 regularization_coefficients=None,
+                 regularization_parameter=0):
+        """FPCABasis constructor
+
+        Args:
+            n_components (int): number of principal components to obtain from
+                functional principal component analysis
+            components_basis (skfda.representation.Basis): the basis in which we
+                want the principal components. Defaults to None. If so, the
+                basis contained in the passed FDataBasis object for the fit
+                function will be used.
+            centering (bool): if True then calculate the mean of the functional
+                data object and center the data first. Defaults to True
+            regularization_parameter (float): this parameter sets the degree of
+                regularization that is desired. Defaults to 0 (no
+                regularization). When this value is large, the resulting
+                principal components tends to be 0.
+
+        """
+        super().__init__(n_components, centering)
+        # basis that we want to use for the principal components
+        self.components_basis = components_basis
+        # lambda in the regularization / penalization process
+        self.regularization_parameter = regularization_parameter
+        self.regularization_derivative_degree = regularization_derivative_degree
```

Comment:
After #189, we will favour receiving a single parameter which will be used to initialize a LinearDifferentialOperator (if it is not one already).